Tasks/PackerBuildV0/DefaultTemplates/custom.managed.linux.template.json

```diff
@@ -0,0 +1,68 @@
+{
+  "variables": {
+    "client_id": "{{env `client_id`}}",
+    "client_secret": "{{env `client_secret`}}",
+    "subscription_id": "{{env `subscription_id`}}",
+    "tenant_id": "{{env `tenant_id`}}",
+    "object_id": "{{env `object_id`}}",
+    "resource_group": "{{env `resource_group`}}",
+    "storage_account": "{{env `storage_account`}}",
+    "image_url": "{{env `image_url`}}",
+    "location": "{{env `location`}}",
+    "script_relative_path": "{{env `script_relative_path`}}",
+    "script_arguments": "{{env `script_arguments`}}",
+    "package_path": "{{env `package_path`}}",
+    "package_name": "{{env `package_name`}}",
+    "capture_name_prefix": "{{env `capture_name_prefix`}}",
+    "skip_clean": "{{env `skip_clean`}}",
+    "managed_image_name":  "{{env `managed_image_name`}}"
+  },
+  "builders": [{
+        "type": "azure-arm",
+        "client_id": "{{user `client_id`}}",
+        "client_secret": "{{user `client_secret`}}",
+        "managed_image_resource_group_name": "{{user `resource_group`}}",
+        "managed_image_name": "{{user `managed_image_name`}}",
+        "subscription_id": "{{user `subscription_id`}}",
+        "tenant_id": "{{user `tenant_id`}}",
+        "object_id": "{{user `object_id`}}",
+                
+        "os_type": "Linux",
+        "image_url": "{{user `image_url`}}",
+        "location": "{{user `location`}}",
+        "vm_size": "Standard_D3_v2"
+
+    }],
+    "provisioners": [{
+        "type": "shell",
+        "execute_command": "chmod +x {{ .Path }}; {{ .Vars }} sudo -E sh '{{ .Path }}'",
+        "inline": [
+            "rm -rf /deployTemp",
+            "mkdir /deployTemp",
+            "chmod 777 /deployTemp"
+        ]
+    },
+    {
+        "type": "file",
+        "source": "{{user `package_path`}}",
+        "destination": "/deployTemp/{{user `package_name`}}"
+    },
+    {
+        "type": "shell",
+        "execute_command": "chmod +x {{ .Path }}; {{ .Vars }} sudo -E sh '{{ .Path }}'",
+        "inline": [
+            "cd /deployTemp",
+            "ls",
+            "chmod +x /deployTemp/{{user `package_name`}}/{{user `script_relative_path`}}",
+            "/deployTemp/{{user `package_name`}}/{{user `script_relative_path`}} {{user `script_arguments`}}"
```

Comment:
do you need to source as well?

Reply:
Have added the source to the custom templates. However I tested with my own custom managed and unmanaged templates.
unmanaged

```json
{
  "variables": {
    "tenant_id": "{{env `ARM_TENANT_ID`}}",
    "client_id": "{{env `ARM_CLIENT_ID`}}",
    "client_secret": "{{env `ARM_CLIENT_SECRET`}}",
    "subscription_id": "{{env `ARM_SUBSCRIPTION_ID`}}",
    "managed_image_resource_group_name": "myresourcegroupname",
    "managed_image_name": "mymanagedlinuximage"
  },
  "builders": [
    {
      "type": "azure-arm",
      "client_id": "{{user `client_id`}}",
      "client_secret": "{{user `client_secret`}}",
      "managed_image_resource_group_name": "{{user `managed_image_resource_group_name`}}",
      "managed_image_name": "{{user `managed_image_name`}}",
      "subscription_id": "{{user `subscription_id`}}",
      "tenant_id": "{{user `tenant_id`}}",
      "os_type": "Linux",
      "image_publisher": "Canonical",
      "image_offer": "UbuntuServer",
      "image_sku": "16.04.0-LTS",
      "location": "West US",
      "vm_size": "Standard_A2"
    }
  ],
  "provisioners": [
    {
      "execute_command": "chmod +x {{ .Path }}; {{ .Vars }} sudo -E sh '{{ .Path }}'",
      "inline": [
        "apt-get update",
        "apt-get upgrade -y",
        "apt-get install ansible -y"
      ],
      "inline_shebang": "/bin/sh -x",
      "type": "shell"
    },
    {
      "type": "file",
      "source": "{{ template_dir }}/../../data.php",
      "destination": "/tmp/data.php"
    },
    {
      "type": "file",
      "source": "{{ template_dir }}/../../drill.html",
      "destination": "/tmp/drill.html"
    },
    {
      "type": "file",
      "source": "{{ template_dir }}/../../index.html",
      "destination": "/tmp/index.html"
    },
    {
      "type": "file",
      "source": "{{ template_dir }}/../../dist",
      "destination": "/tmp/dist"
    },
    {
      "type": "ansible-local",
      "playbook_file": "{{ template_dir }}/phpapp-packer-ansible-provisioner.yml"
    },
    {
      "execute_command": "chmod +x {{ .Path }}; {{ .Vars }} sudo -E sh '{{ .Path }}'",
      "inline": [
        "/usr/sbin/waagent -force -deprovision+user && export HISTSIZE=0 && sync"
      ],
      "inline_shebang": "/bin/sh -x",
      "type": "shell"
    }
  ]
}
```